GANDLF/metrics/metric_calculators.py

```diff
@@ -0,0 +1,88 @@
+import torch
+from copy import deepcopy
+from GANDLF.metrics import get_metrics
+from abc import ABC, abstractmethod
+
+
+class AbstractMetricCalculator(ABC):
+    def __init__(self, params: dict):
+        super().__init__()
+        self.params = deepcopy(params)
+        self._initialize_metrics_dict()
+
+    def _initialize_metrics_dict(self):
+        self.metrics_calculators = get_metrics(self.params)
+
+    def _process_metric_value(self, metric_value: torch.Tensor):
+        if metric_value.dim() == 0:
+            return metric_value.item()
+        else:
+            return metric_value.tolist()
+
+    @abstractmethod
+    def __call__(
+        self, prediction: torch.Tensor, target: torch.Tensor, *args
+    ) -> torch.Tensor:
+        pass
+
+
+class MetricCalculatorStdnet(AbstractMetricCalculator):
```

Comment:
Same here (it should be `SDNet`, perhaps?).

Reply:
yes